### PR TITLE
Loading of external components

### PR DESCRIPTION
**Pull request by ybznek**
I've decided to split part of my changes to loading external components.

This should make working such references
```
xlink:href="file:xccdf.sds.xml#scap_1_comp_xccdf.xml"
```

```
xlink:href="file:xccdf.xml"
```

### CODE REVIEW

**Comment by mpreisler** [comment]
@openscap-jenkins retest this please, one of the fails is a file probe

The other fail is `test_ds.sh` and that looks like a legit issue.

File: src/DS/sds.c [hunk 1]
```diff
@@ -261,15 +310,44 @@ int ds_sds_dump_component_ref_as(const xmlNodePtr component_ref, struct ds_sds_s
...
+
+
+	} else {
+		oscap_seterr(OSCAP_EFAMILY_XML, "Unsupported type of xlink:href attribute on given component-ref - '%s'.", xlink_href);
+		xmlFree(cref_id);
+		xmlFree(xlink_href);
+		return 0;
+	}
+
+    char* filename_cpy = oscap_sprintf("./%s", relative_filepath);
```
Comment (jan-cerny) [inline src/DS/sds.c]: Please use tabs instead of spaces.

**Comment by ybznek** [comment]
I have rebased to fix whitespaces issue and missing "subdir" in Makefile.am for tests

**Comment by jan-cerny** [comment]
@ybznek I have tested your branch and it works well for me. 
I very appreciate the fact that you have splitted your work to more separate pull requests. That makes the review much more easier and work gets faster.

Overall the commits LGTM. Great job!

Just please next time try give some more love to the coding style. I know that the coding style does not affect the functionality of the program at all, but following it helps others to read the code and orient in the code better. Thank you.

**Comment by ybznek** [comment]
@jan-cerny 
I'have divided code into more functions in next pull requests - I don't like very long functions in oscap.
Can you please be more explicit about problems with my coding style?

**Pull request merged**